packages/lib-utils/src/k8s/hooks/useK8sWatchResource.test.ts

```diff
@@ -0,0 +1,179 @@
+      K8sModelCommon,
+      boolean,
+    ]);
+
+    const { result } = renderHook(() => useK8sWatchResource(null));
+    const [data, loaded, error] = result.current;
+
+    expect(data).toBeUndefined();
+    expect(loaded).toEqual(true);
+    expect(error).toBeUndefined();
+  });
+
+  test('should return "loaded" as false when models have not loaded', () => {
+    const haveModelsLoaded = false;
+    (useModelsLoaded as jest.Mock).mockReturnValue(haveModelsLoaded);
+    (useSelector as jest.Mock).mockReturnValue(null); // get resourceK8s
+    (useDeepCompareMemoize as jest.Mock).mockReturnValue(watchedResourceMock);
+    (useK8sModel as jest.Mock).mockReturnValue([undefinedModelMock, false] as [
+      K8sModelCommon,
+      boolean,
+    ]);
+
+    const { result } = renderHook(() => useK8sWatchResource(watchedResourceMock));
+    const [data, loaded, error] = result.current;
+
+    expect(data).toMatchObject({});
+    expect(loaded).toEqual(false);
+    expect(error).toBeUndefined();
+  });
+
+  test('should return specific error if the model for the watched resource does not exist', () => {
+    (useSelector as jest.Mock).mockReturnValue(null); // get resourceK8s
+    (useModelsLoaded as jest.Mock).mockReturnValue(true);
+    (useDeepCompareMemoize as jest.Mock).mockReturnValue(watchedResourceMock);
+    (useK8sModel as jest.Mock).mockReturnValue([undefinedModelMock, false] as [
+      K8sModelCommon,
+      boolean,
+    ]);
+
+    const { result } = renderHook(() => useK8sWatchResource(watchedResourceMock));
+    const [data, loaded, error] = result.current;
+
+    expect(data).toMatchObject({});
+    expect(loaded).toEqual(true);
+    expect((error as NoModelError).message).toEqual('Model does not exist');
+  });
+
+  test('should return data for the watched resource', () => {
+    const mockWatchData: WatchData = {
+      id: 'appstudio.redhat.com~v1alpha1~Application---{"ns":"test-ns","name":"test"}',
+      action: jest.fn(),
+    };
+    const payload = {
+      data: resourceDataMock,
+      loaded: true,
+      loadError: '',
+    };
+    const reduxIdPayload: ImmutableMap<string, unknown> = ImmutableMap(payload);
+    (useSelector as jest.Mock).mockReturnValue(reduxIdPayload); // get resourceK8s
+    (getReduxData as jest.Mock).mockReturnValue(resourceDataMock);
+    (useModelsLoaded as jest.Mock).mockReturnValue(true);
+    (useDeepCompareMemoize as jest.Mock).mockReturnValue(watchedResourceMock);
+    (useK8sModel as jest.Mock).mockReturnValue([resourceModelMock, false] as [
+      K8sModelCommon,
+      boolean,
+    ]);
+    (getWatchData as jest.Mock).mockReturnValue(mockWatchData);
+
+    const { result } = renderHook(() => useK8sWatchResource(watchedResourceMock));
+    const [data, loaded, error] = result.current;
+
+    expect(useSelector as jest.Mock).toHaveBeenCalled();
+    const resourceData = data as K8sResourceCommon;
+    expect(resourceData?.metadata?.creationTimestamp).toEqual('2022-04-29T13:41:21Z');
```

Comment:
@vidyanambiar we may want to consider comparing more than just the timestamps to cover all of the bases and to also avoid hard-coding the right hand side of this assertion.

We could achieve this through the [objectContaining](https://jestjs.io/docs/expect#expectobjectcontainingobject) assertion perhaps.

Reply:
Otherwise if we update the creation timestamp on line 39 this test and any other tests with the hard-coded value of `2022-04-29T13:41:21Z` will fail.